XXX: just a sample patch illustratin how to remove rosbuild for ros-groovy
XXX: this patch should be integrated upstream (either this plain cmake or by using
XXX: ros-catkin) before inclusion in robotpkg

--- CMakeLists.txt.orig	2013-07-10 13:40:55.435865605 +0200
+++ CMakeLists.txt	2013-07-10 13:41:09.599819977 +0200
@@ -1,5 +1,5 @@
-cmake_minimum_required(VERSION 2.4.6)
-include($ENV{ROS_ROOT}/core/rosbuild/rosbuild.cmake)
+cmake_minimum_required(VERSION 2.6)
+cmake_policy(SET CMP0003 NEW)
 
 INCLUDE(cmake/base.cmake)
 INCLUDE(cmake/dist.cmake)
@@ -23,30 +23,30 @@
 
 # end of the partial extraction from SETUP_PROJECT
 
-
-#set(ROS_BUILD_TYPE RelWithDebInfo)
-set(ROS_BUILD_TYPE Debug)
-
-rosbuild_init()
-
 #set the default path for built executables to the "bin" directory
 set(EXECUTABLE_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/bin)
 #set the default path for built libraries to the "lib" directory
 set(LIBRARY_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/lib)
 
 # Search for dependencies.
-ADD_REQUIRED_DEPENDENCY("jrl-dynamics")
-
-rosbuild_add_boost_directories()
-
-# Make sure Boost.Filesystem v2 is used.
-add_definitions(-DBOOST_FILESYSTEM_VERSION=2)
-
-rosbuild_add_library(${PROJECT_NAME} src/parser.cc)
-rosbuild_link_boost(${PROJECT_NAME} filesystem system)
-PKG_CONFIG_USE_DEPENDENCY(${PROJECT_NAME} jrl-dynamics)
+find_package(PkgConfig)
+pkg_check_modules(DEPS bullet jrl-dynamics jrl-mal resource_retriever>=1.9 urdf>=1.9)
+find_package(Boost 1.36.0 REQUIRED filesystem system)
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
+include_directories(${Boost_INCLUDE_DIRS})
+include_directories(${DEPS_INCLUDE_DIRS})
+link_directories(${DEPS_LIBRARY_DIRS})
+
+# targets
+add_library(${PROJECT_NAME} SHARED src/parser.cc)
+target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES} ${DEPS_LIBRARIES})
+install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION lib)
 
-rosbuild_add_executable(display-robot examples/display-robot.cc)
-target_link_libraries(display-robot ${PROJECT_NAME})
+add_executable(display-robot examples/display-robot.cc)
+target_link_libraries(display-robot ${PROJECT_NAME} ${DEPS_LIBRARIES})
+install(TARGETS display-robot RUNTIME DESTINATION bin)
+
+install(DIRECTORY include/ DESTINATION include FILES_MATCHING PATTERN "*.hh")
 
 add_subdirectory(tests)
